src/components/TableComplex.tsx

```diff
@@ -0,0 +1,150 @@
+import {
+  ColumnDef,
+  flexRender,
+  getCoreRowModel,
+  getFilteredRowModel,
+  getPaginationRowModel,
+  Table,
+  useReactTable
+} from '@tanstack/react-table';
+import classnames from 'classnames';
+import { Pagination, PaginationProps } from './Pagination';
+import './Table.less';
+import { TableFilter as Filter } from './TableFilter';
+
+export interface TableComplexOptions {
+  isFilterable?: boolean;
+  isStriped?: boolean;
+  pageSize?: number;
+  isPaginated?: boolean;
+  isResponsive?: boolean;
+}
+
+interface TableComplexProps {
+  /** Source data */
+  data: object[];
+  /** Column definitions (react-table) */
+  columns: ColumnDef<any>[];
+  /** Table description */
+  caption?: React.ReactElement | string;
```

Comment:
Curious: Do you think it's worth combining `React.ReactElement | string` into [`React.ReactNode`](https://reactnative.dev/docs/react-node)?